lib/active_merchant/billing/gateways/quickpay/quickpay_v10.rb

```diff
@@ -182,13 +197,24 @@ def parse(body)
 
         def successful?(response)
           has_error    = response['errors']
-          invalid_code = (response.key?('qp_status_code') and response['qp_status_code'] != "20000")
+          invalid_code = invalid_operation_code(response)
 
           !(has_error || invalid_code)
         end
 
         def message_from(success, response)
-          success ? 'OK' : (response['message'] || response['qp_status_msg'])
+          success ? 'OK' : (response['message'] || invalid_operation_message(response))
+        end
+        
+        def invalid_operation_code response
```

Comment:
Although ruby allows you to omit braces on method definitions, they should still be there. Please add brackets `def invalid_operation_code(response)`. Same thing for `def invalid_operation_message(response)`.